### ISSUE TITLE
Infinite warnings while using `mask`

### ISSUE THREAD

**Issue by ayushpatnaikgit** [rafaqz/Rasters.jl#347]
I am getting infinite warnings while using `Rasters.mask` 
```julia
using Rasters, Shapefile, DataFrames
raster = Raster("DATA/Harmonized_DN_NTL_2020_simVIIRS.tif")
shp = Shapefile.Table("DATA/world_shapefile/in_primary_countries.shp") |> DataFrame
japan_index = findall( x -> x == "Japan", shp.name)
japan = crop(raster; to = shp.geometry[japan_index])
japan_masked = mask(japan, with = shp.geometry[japan_index]) 
```
```julia
┌ Warning: (Band,) dims were not found in object
└ @ DimensionalData.Dimensions ~/.julia/packages/DimensionalData/yMNeq/src/Dimensions/primitives.jl:648
┌ Warning: (Band,) dims were not found in object
└ @ DimensionalData.Dimensions ~/.julia/packages/DimensionalData/yMNeq/src/Dimensions/primitives.jl:648
┌ Warning: (Band,) dims were not found in object
└ @ DimensionalData.Dimensions ~/.julia/packages/DimensionalData/yMNeq/src/Dimensions/primitives.jl:648
┌ Warning: (Band,) dims were not found in object
└ @ DimensionalData.Dimensions ~/.julia/packages/DimensionalData/yMNeq/src/Dimensions/primitives.jl:648
```

I am also getting these while using `Rasters.zonal`
```julia
zonal(sum, raster; of=shp.geometry)
```

The performance also seems bad and variable. On some occasions, it takes 30 seconds and on some, it takes up to an hour. I am unable to replicate that, perhaps it's related to my memory usage. 

For replication, the tif file can be obtained [here](https://github.com/xKDR/datascience-tutorials/blob/main/DATA/Harmonized_DN_NTL_2020_simVIIRS.tif). And the Shapefile can be obtained [here](https://github.com/xKDR/datascience-tutorials/tree/main/DATA/world_shapefile). 

I've also tried the same with a [different tif file](https://github.com/xKDR/datascience-tutorials/blob/main/DATA/MODISAQ/MYDAL2_M_AER_OD_2020-01.tiff). 

PS:

I am making a tutorial on using Rasters, and this is something I found while making the Jupyter notebook for it.  
https://github.com/xKDR/datascience-tutorials/blob/main/rasters.ipynb

**Comment by rafaqz** [comment]
I think this is already fixed on master, and the 0.4.5 patch version should be registered in the next half hour.

Good work on the notebook! will this be in a blog post or something similar?

**Comment by ayushpatnaikgit** [comment]
Is it fixed! The speed is spectacular.

I've been teaching raster analysis at a few places, so I thought I'll make a notebook + video and circulate it.
